File: weather_impact.js
const XLSX = require('xlsx');
const fs = require('fs');

// Read the Excel file
console.log('Reading McDonald\'s restaurant data...');
const workbook = XLSX.readFile('McDonald restaurants.xlsx');
const sheetName = workbook.SheetNames[0];
const worksheet = workbook.Sheets[sheetName];
const data = XLSX.utils.sheet_to_json(worksheet);

// Display structure
console.log('\nFile structure:');
console.log('First row:', data[0]);
console.log('\nColumns:', Object.keys(data[0]));
console.log(`\nTotal restaurants: ${data.length}`);

// Check for coordinates
const hasCoordinates = data[0].hasOwnProperty('latitude') ||
                       data[0].hasOwnProperty('lat') ||
                       data[0].hasOwnProperty('Latitude');

console.log(hasCoordinates ? '\n✓ Coordinates found' : '\n✗ No coordinates - will need geocoding');

File: weather_impact.py
import pandas as pd
import requests
from datetime import datetime, timedelta
import time

# Read the Excel file
print("Reading McDonald's restaurant data...")
df = pd.read_excel('McDonald restaurants.xlsx')

# Display the structure
print("\nFile structure:")
print(df.head())
print("\nColumns:", df.columns.tolist())
print(f"\nTotal restaurants: {len(df)}")

# Check if we have coordinates or need to geocode
if 'latitude' in df.columns or 'lat' in df.columns:
    print("\n✓ Coordinates found in file")
else:
    print("\n✗ No coordinates found - will need to geocode addresses")
